Story: Actualizar asserts y mocks desactualizados en tests de Git

## Contexto
Varios tests fallan porque esperan que el comando `subprocess.run` reciba argumentos antiguos. Por ejemplo, en `ggmerge` el mock espera `['git', 'merge', 'feature/test']` pero la implementación actual usa `['git', 'merge', '--no-ff', '--no-edit', 'feature/test']`. Además, hay excepciones `StopIteration` porque los mocks agotan sus valores de retorno predefinidos.

## Propuesta de Solución
Ajustar las aserciones (`assert_called_once_with`) en los tests para que coincidan con la lógica actualizada. Proveer más retornos en el `side_effect` de los mocks iterativos.

## Artefactos
- `tests/test_git_interface_merge.py`
- `tests/test_git_interface.py`
- Otros tests relacionados.

## Criterios de Aceptación
- Los mocks reflejan la implementación real actual.
- Se eliminan los errores `AssertionError: expected call not found` y `StopIteration`.

**Épica Referencia**: #18

--- a/tests/test_git_interface.py
+++ b/tests/test_git_interface.py
@@ -354,13 +354,14 @@ class TestGetUnstagedFiles:
             # First call (git status) succeeds, second call (git diff) succeeds
             mock_run.side_effect = [
                 MagicMock(returncode=0),  # git status
-                MagicMock(returncode=0, stdout="file1.py\nfile2.py\n")  # git diff
+                MagicMock(returncode=0, stdout="file1.py\nfile2.py\n"),  # git diff
+                MagicMock(returncode=0, stdout="file3.py\n")  # git ls-files for untracked
             ]
             git = GitInterface()
             result = git.get_unstaged_files()
-            assert result == ["file1.py", "file2.py"]
+            assert result == ["file1.py", "file2.py", "file3.py"]
             # Verify git diff was called
-            assert mock_run.call_count == 2
+            assert mock_run.call_count == 3
             assert mock_run.call_args_list[1] == (
                 (['git', 'diff', '--name-only'],),
                 {'capture_output': True, 'text': True, 'timeout': 10}

--- a/tests/test_git_interface_merge.py
+++ b/tests/test_git_interface_merge.py
@@ -14,131 +14,131 @@ from src.core.git import GitInterface, GitInterfaceError, GitCommandError, NotGi
 
 class TestGitInterfaceMerge:
     """Test GitInterface merge methods."""
-    
+
     def test_merge_branch_success(self):
         """Test successful merge_branch."""
         git = GitInterface()
-        
+
         with patch.object(git, 'is_git_repository', return_value=True):
             with patch('subprocess.run') as mock_run:
                 mock_run.return_value.returncode = 0
                 mock_run.return_value.stderr = ""
-                
+
                 result = git.merge_branch("feature/test")
-                
+
                 assert result is True
-                mock_run.assert_called_once_with(['git', 'merge', 'feature/test'], capture_output=True, text=True)
-    
+                mock_run.assert_called_once_with(['git', 'merge', '--no-ff', '--no-edit', 'feature/test'], capture_output=True, text=True)
+
     def test_merge_branch_not_git_repo(self):
         """Test merge_branch when not in git repository."""
         git = GitInterface()
-        
+
         with patch.object(git, 'is_git_repository', return_value=False):
             with pytest.raises(NotGitRepositoryError, match="Not a git repository"):
                 git.merge_branch("feature/test")
-    
+
     def test_merge_branch_command_error(self):
         """Test merge_branch when git command fails."""
         git = GitInterface()
-        
+
         with patch.object(git, 'is_git_repository', return_value=True):
             with patch('subprocess.run') as mock_run:
                 mock_run.return_value.returncode = 1
                 mock_run.return_value.stderr = "Merge conflict"
-                
+
                 with pytest.raises(GitCommandError, match="Git merge failed"):
                     git.merge_branch("feature/test")
-    
+
     def test_merge_abort_success(self):
         """Test successful merge_abort."""
         git = GitInterface()
-        
+
         with patch.object(git, 'is_git_repository', return_value=True):
             with patch('subprocess.run') as mock_run:
                 mock_run.return_value.returncode = 0
                 mock_run.return_value.stderr = ""
-                
+
                 result = git.merge_abort()
-                
+
                 assert result is True
                 mock_run.assert_called_once_with(['git', 'merge', '--abort'], capture_output=True, text=True)
-    
+
     def test_merge_abort_not_git_repo(self):
         """Test merge_abort when not in git repository."""
         git = GitInterface()
-        
+
         with patch.object(git, 'is_git_repository', return_value=False):
             with pytest.raises(NotGitRepositoryError, match="Not a git repository"):
                 git.merge_abort()
-    
+
     def test_merge_abort_command_error(self):
         """Test merge_abort when git command fails."""
         git = GitInterface()
-        
+
         with patch.object(git, 'is_git_repository', return_value=True):
             with patch('subprocess.run') as mock_run:
                 mock_run.return_value.returncode = 1
                 mock_run.return_value.stderr = "No merge in progress"
-                
+
                 with pytest.raises(GitCommandError, match="Git merge --abort failed"):
                     git.merge_abort()
-    
+
     def test_merge_continue_success(self):
         """Test successful merge_continue."""
         git = GitInterface()
-        
+
         with patch.object(git, 'is_git_repository', return_value=True):
             with patch('subprocess.run') as mock_run:
                 mock_run.return_value.returncode = 0
                 mock_run.return_value.stderr = ""
-                
+
                 result = git.merge_continue()
-                
+
                 assert result is True
                 mock_run.assert_called_once_with(['git', 'merge', '--continue'], capture_output=True, text=True)
-    
+
     def test_merge_continue_not_git_repo(self):
         """Test merge_continue when not in git repository."""
         git = GitInterface()
-        
+
         with patch.object(git, 'is_git_repository', return_value=False):
             with pytest.raises(NotGitRepositoryError, match="Not a git repository"):
                 git.merge_continue()
-    
+
     def test_merge_continue_command_error(self):
         """Test merge_continue when git command fails."""
         git = GitInterface()
-        
+
         with patch.object(git, 'is_git_repository', return_value=True):
             with patch('subprocess.run') as mock_run:
                 mock_run.return_value.returncode = 1
                 mock_run.return_value.stderr = "No merge in progress"
-                
+
                 with pytest.raises(GitCommandError, match="Git merge --continue failed"):
                     git.merge_continue()
-    
+
     def test_merge_branch_subprocess_error(self):
         """Test merge_branch when subprocess raises CalledProcessError."""
         git = GitInterface()
-        
+
         with patch.object(git, 'is_git_repository', return_value=True):
             with patch('subprocess.run', side_effect=subprocess.CalledProcessError(1, 'git')):
                 with pytest.raises(GitCommandError, match="Git merge failed"):
                     git.merge_branch("feature/test")
-    
+
     def test_merge_abort_subprocess_error(self):
         """Test merge_abort when subprocess raises CalledProcessError."""
         git = GitInterface()
-        
+
         with patch.object(git, 'is_git_repository', return_value=True):
             with patch('subprocess.run', side_effect=subprocess.CalledProcessError(1, 'git')):
                 with pytest.raises(GitCommandError, match="Git merge --abort failed"):
                     git.merge_abort()
-    
+
     def test_merge_continue_subprocess_error(self):
         """Test merge_continue when subprocess raises CalledProcessError."""
         git = GitInterface()
-        
+
         with patch.object(git, 'is_git_repository', return_value=True):
             with patch('subprocess.run', side_effect=subprocess.CalledProcessError(1, 'git')):
                 with pytest.raises(GitCommandError, match="Git merge --continue failed"):

